### ISSUE TITLE
3 sprints exist (WIP)

### ISSUE THREAD

**Issue by S-B99** [S-B99/KAMI#22]
Middle one works

This is (probably) due to removing old features and adding new ones recently
![Screenshot_20191018_120125](https://user-images.githubusercontent.com/17222512/67109657-035c8580-f19f-11e9-9a0c-cabb46de5d36.png)

**Comment by S-B99** [comment]
Duplicated by #125, watch that for a solution if I don't find one first.

**Comment by awfol** [comment]
not sure if it matters but I'm pretty sure I can also remember having 2 sprint modules

**Comment by S-B99** [comment]
I confirmed the other day on discord the middle Sprint is the real one. 
Hacky solution to fix it if I can't figure out why it's there is to just hide the other 2

**Comment by S-B99** [comment]
Debugging this is going to be super fun and I totally have a lot of time on my hands :upside_down_face:

**Comment by S-B99** [comment]
Help found, going to test and implement later today

**Comment by cookiedragon234** [comment]
Discussed this in DMs with @S-B99 , after looking around I found this code Module.java#L43 where if an annotation cant be found for a hack it defaults to Sprint, giving hacks without a name the sprint name + description + category. (Tip for debugging in the future, I just put Sprint into the search bar).

My suggested fix is replacing the function at the mentioned lines with:
```Java
if(getClass().isAnnotationPresent(Info.class))
{
	return getClass().getAnnotation(Info.class);
}
throw new IllegalStateException("No Annotation on class " + this.getClass().getCanonicalName()
```

**Comment by blockparole** [comment]
I implemented this [here](https://github.com/zeroeightysix/KAMI/pull/133), works fine for me :+1:

**Comment by S-B99** [comment]
Thank you, I'm going to test what the broken modules really are and why toggle doesn't toggle the correct sprint and then implement it

**Comment by zeroeightysix** [comment]
> where if an annotation cant be found for a hack it defaults to Sprint

oops

**Comment by S-B99** [comment]
😳

**Comment by S-B99** [comment]
Debugging done

This is why there's 3 sprints

I will add a proper debugging module, hide it to the user and replace sprint with debugging in Modules.java

https://github.com/S-B99/KAMI/blob/features-master/src/main/java/me/zeroeightsix/kami/module/modules/misc/Carpenter.java

And portal tracers

**Comment by S-B99** [comment]
nevermind just delete the module lol

@zeroeightysix why did you comment out the 2 modules annotations and add this code instead of just deleting it smh

**Comment by S-B99** [comment]
FIxed in commit 083e93546b37cb8a025134e479c7e0baecb1c485
